src/components/modal/ModalCreateBooks.tsx

```diff
@@ -0,0 +1,188 @@
+import { FormEvent, useEffect, useState } from 'react'
+
+import axios from 'axios'
+import Select, { MultiValue } from 'react-select'
+import ReactDOM from 'react-dom'
+import './ModalCreateBooks.css'
+import { GrClose as XIconForModal } from 'react-icons/gr'
+import { BiBookAdd as BookAddIcon } from 'react-icons/bi'
+
+import BookBody from '../../model/Book'
+import BookService from '../../services/BookService'
+import Author from '../../model/Author'
+import AuthorService from '../../services/AuthorService'
+import placeholder from '../../assets/placeholder/placeholderForBook.png'
+
+
+interface Modal {
+  open: boolean,
+  onClose: () => void
+}
+
+
+const poratlDiv = document.getElementById('portal') as HTMLElement
+const ModalCreateBooks = ({ open , onClose } : Modal) => {
+
+  const [ showAddAuthor, setShowAddAuthor ] = useState(true)
+  const [ showAddButton, setShowAddButton ] = useState(true)
+  const [ authors, setAtuhors ] = useState([])
+  const [ fileImg, setFileImg ] = useState<Blob>(new Blob())
+  const [ coverImg, setCoverImg ] = useState('')
+  const [ booksData, setBooksData ] = useState<BookBody>({
+    Id: 0,
+    Title: '',
+    Description: '',
+    Isbn: '',
+    Quantity: 0,
+    Cover: fileImg,
+    PublishDate: '',
+    AuthorIds: []
+  })
+
+  const [ authorData, setAuthorData ] = useState<Author>({
+    Id: 0,
+    FirstName: '',
+    LastName: ''
+  })
+
+
+  const handleImgUpload = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
+    if (currentTarget.files) {
+      const files = currentTarget.files
+      const reader = new FileReader()
+      if (files) {
```

Comment:
Why do you need check for filles if you have already checked currentTarget.files and assigned it's value to files?

Reply:
I fix this.